### PR TITLE
Add new BTM battery dispatch method for self-consumption

### PR DESCRIPTION
**Pull request by janinefreeman**
This pull request and the complementary one in SAM add a new battery dispatch methodology for behind-the-meter PV+Battery and BTM battery hybrid systems that allow the battery to be dispatched to maximize self-consumption, which can also be termed "24/7 carbon-free energy" for larger industrial loads. Behind the scenes, this new dispatch option uses the grid power target option with the entire timeseries set to 0, but it is a new dispatch option to enable this functionality for PySAM users as well. Added a few new tests to go along with the new dispatch method.

Remaining to be done: adding outputs to help users understand how well this dispatch method is achieving its target (zero grid power at each timestep)

### CODE REVIEW

**Comment by brtietz** [comment]
Test failure here likely due to https://github.com/NREL/ssc/issues/1026. I propose updating the tolerance for the test for now and then lowering it again once we get to that issue.

File: shared/lib_battery_dispatch.h [hunk 1]
```diff
@@ -57,7 +57,7 @@ class dispatch_t
 public:
 
 	enum FOM_MODES { FOM_AUTOMATED_ECONOMIC, FOM_PV_SMOOTHING, FOM_CUSTOM_DISPATCH, FOM_MANUAL };
-	enum BTM_MODES { PEAK_SHAVING, MAINTAIN_TARGET, CUSTOM_DISPATCH, MANUAL, FORECAST };
+	enum BTM_MODES { PEAK_SHAVING, MAINTAIN_TARGET, CUSTOM_DISPATCH, MANUAL, FORECAST, CARBON_FREE };
```
Comment (brtietz) [inline shared/lib_battery_dispatch.h:60]: I hate to slow things down, but having it be CARBON_FREE in the code and self consumption in the UI is going to set us up for confusion later. Mind fixing it to match the UI?

File: test/ssc_test/cmod_battery_pvsamv1_test.cpp [hunk 2]
```diff
@@ -383,7 +388,12 @@ TEST_F(CMPvsamv1BatteryIntegration_cmod_pvsamv1, ResidentialDCBatteryModelIntegr
             pairs["batt_dispatch_wf_forecast_choice"] = 0;
             pairs["batt_dispatch_load_forecast_choice"] = 0;
             break;
-
+        case 3:
+            // Input grid power targets
```
Comment (brtietz) [inline test/ssc_test/cmod_battery_pvsamv1_test.cpp:392]: Update comment to match the correct name, please.

**Review by brtietz — CHANGES_REQUESTED** [review]
I think naming consistency is worth the time here.

Line 426 is where the tolerance needs to be updated, maybe go with 1 for now?

File: shared/lib_battery_dispatch.h (hunk 1 above)
Comment (janinefreeman) [reply shared/lib_battery_dispatch.h:60]: Ha, no you're absolutely right. I had that on my list, will change now.

File: test/ssc_test/cmod_battery_pvsamv1_test.cpp (hunk 2 above)
Comment (janinefreeman) [reply test/ssc_test/cmod_battery_pvsamv1_test.cpp:392]: Done

**Review by brtietz — APPROVED** [review]
Thanks for the code cleanup!

**Pull request merged**